apps/frontend/src/features/graph/AddConsentButton.tsx

```diff
@@ -0,0 +1,75 @@
+import { FC, useEffect, useState } from "react";
+import { toast } from "sonner";
+import { storage } from "@/lib/storage";
+import { logClientEvent } from "@/lib/frontend/metrics";
+import { AppButton } from "@/components/ui/Button";
+import { createToggleSocialGraphBackup } from "@/lib/backup";
+import { saveBackupAndUpdateStorage } from "@/lib/storage/localStorage/utils";
+
+interface AddGraphConsentButtonProps {
+  // If you need to open in new tab vs same window
+  openInNewTab?: boolean;
+  // Optional custom button text
+  buttonText?: string;
+  // Optional className for additional styling
+  className?: string;
+}
+
+export const AddGraphConsentButton: FC<AddGraphConsentButtonProps> = ({
+    className,
+  }) => {
+
+  const [buttonText, setButtonText] =
+    useState("Test");
+  const [updatedSetting, setUpdatedSetting] =
+    useState(false);
+
+  useEffect(() => {
+    const checkSetting = async () => {
+      const { user } = await storage.getUserAndSession();
+      setButtonText(user.tapGraphEnabled ? 'Disable Graph' : 'Enable Graph');
+      setUpdatedSetting(false);
+    }
+    checkSetting();
+  }, [updatedSetting]);
+
+  const handleGraphConsentSetup = async () => {
+    logClientEvent("toggle-tap-graph", {});
+
+    const { user, session } = await storage.getUserAndSession();
```

Comment:
this entire function should probably be in storage, i.e. storage.toggleSocialGraphConsent